retworkx/__init__.py

```diff
@@ -81,3 +82,356 @@ class PyDAG(PyDiGraph):
     :meth:`PyDAG.add_parent` will avoid this overhead.
     """
     pass
+
+
+@functools.singledispatch
+def distance_matrix(graph, parallel_threshold=300):
+    """Get the distance matrix for a graph
+
+    This differs from functions like :func:`~retworkx.floyd_warshall_numpy` in
+    that the edge weight/data payload is not used and each edge is treated as a
+    distance of 1.
+
+    This function is also multithreaded and will run in parallel if the number
+    of nodes in the graph is above the value of ``parallel_threshold`` (it
+    defaults to 300). If the function will be running in parallel the env var
+    ``RAYON_NUM_THREADS`` can be used to adjust how many threads will be used.
+
+    :param graph: The graph to get the distance matrix for, can be either a
+        :class:`~retworkx.PyGraph` or :class:`~retworkx.PyDiGraph`.
+    :param int parallel_threshold: The number of nodes to calculate the
+        the distance matrix in parallel at. It defaults to 300, but this can
+        be tuned
+    :param bool as_undirected: If set to ``True`` the input directed graph
+        will be treat as if each edge was bidirectional/undirected in the
+        output distance matrix.
+
+    :returns: The distance matrix
+    :rtype: numpy.ndarray
+    """
+    raise TypeError("Invalid Input Type %s for graph" % type(graph))
+
+
+@distance_matrix.register(PyDiGraph)
+def _digraph_distance_matrix(graph, parallel_threshold=300):
```

Comment:
I'm guessing this matches the default for `digraph_distance_matrix`?

Reply:
Yeah: https://github.com/Qiskit/retworkx/blob/master/src/lib.rs#L1285-L1287

Reply:
but that does point out I'm missing `as_undirected` here. I'll fix that now

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/241/commits/fa90541e841f0c9957ec2e6783fb0d77c1529da2